data-plane/dispatcher-loom/src/main/java/dev/knative/eventing/kafka/broker/dispatcherloom/LoomKafkaConsumer.java

```diff
@@ -105,10 +107,10 @@ public Future<Void> close() {
                 }
                 taskRunnerThread.interrupt();
                 taskRunnerThread.join();
-                promise.complete();
+                promise.tryComplete();
```

Comment:
is this behavioral change (try complete - instead of "force complete"), due to the loom env?

Reply:
No, It is not a behavioral change, it is only to make sure we do not mark the promise complete if it was already completed.